sound/soc/intel/boards/Kconfig

```diff
@@ -423,7 +423,7 @@ config SND_SOC_INTEL_SKL_HDA_DSP_GENERIC_MACH
 
 endif ## SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC || SND_SOC_SOF_HDA_AUDIO_CODEC
 
-if SND_SOC_SOF_HDA_COMMON || SND_SOC_SOF_BAYTRAIL
+if (SND_SOC_SOF_HDA_COMMON || SND_SOC_SOF_BAYTRAIL) && SND_SOC_SOF_HDA_LINK
```

Comment:
the logic seems wrong.
If you only want to compile for Baytrail, this will prevent this machine driver from being selected.

Reply:
like I said earlier, you should have 

````
if (SND_SOC_SOF_HDA_LINK || SND_SOC_SOF_BAYTRAIL)
    select HDAC_HDMI if SND_SOC_SOF_HDA_LINK
````

Reply:
@plbossart 

```
select HDAC_HDMI if SND_SOC_SOF_HDA_LINK
```
should be 
```
select HDAC_HDMI
```
as we have judged
```
 if (SND_SOC_SOF_HDA_LINK || SND_SOC_SOF_BAYTRAIL)
```
right?

Do you mean we should 
```
if SND_SOC_SOF_BAYTRAIL || SND_SOC_SOF_HDA_LINK
config SND_SOC_INTEL_SOF_RT5682_MACH
	tristate "SOF with rt5682 codec in I2S Mode"
	depends on I2C && ACPI
	depends on (SND_SOC_SOF_HDA_COMMON && (MFD_INTEL_LPSS || COMPILE_TEST)) ||\
		   (SND_SOC_SOF_BAYTRAIL && (X86_INTEL_LPSS || COMPILE_TEST))
	select SND_SOC_RT5682
	select SND_SOC_DMIC
	select SND_SOC_HDAC_HDMI
	help
	   This adds support for ASoC machine driver for SOF platforms
	   with rt5682 codec.
	   Say Y if you have such a device.
	   If unsure select "N".
endif
```